File: scripts/extract/extract_all.py
import json
import sys

base = '/tmp/re9-translate/json/natives/stm/message/dialog/'

files = [
    'dialog_mangm2.msg.23.json',
    'dialog_movie.msg.23.json',
    'dialog_mangua.msg.23.json',
    'dialog_ff2nd.msg.23.json',
    'dialog_underhb.msg.23.json',
    'dialog_battlepl.msg.23.json',
    'dialog_manlc.msg.23.json',
    'dialog_centba.msg.23.json',
    'dialog_mange.msg.23.json',
    'dialog_platform.msg.23.json',
    'dialog_centuc.msg.23.json',
    'dialog_centhw.msg.23.json',
    'dialog_ending.msg.23.json',
    'dialog_reactionspl.msg.23.json',
]

for fname in files:
    with open(base + fname) as f:
        data = json.load(f)

    print(f"=== {fname} ===")
    for entry in data['entries']:
        en_text = entry.get('strings', {}).get('en', '')
        if en_text and en_text != '<Sound Team ID>' and en_text.strip():
            print(f'{entry["name"]}|{en_text}')
    print()

File: scripts/extract/extract-all.mjs
import { readFileSync, writeFileSync, mkdirSync } from "fs";
import { execSync } from "child_process";
import { decodeMsg } from "remsg";
import { dirname } from "path";

const base = "/mnt/c/Users/henri/Documents/RE-Requiem-CZ";
const outDir = "/tmp/re9-translate/json";

// Find all .msg.23 files
const files = execSync(`find "${base}" -name "*.msg.23"`)
  .toString()
  .trim()
  .split("\n");

let totalStrings = 0;
let totalChars = 0;
const summary = [];

for (const file of files) {
  try {
    const data = readFileSync(file);
    const json = decodeMsg(data);

    const relPath = file.replace(base + "/", "");
    const outPath = outDir + "/" + relPath + ".json";
    mkdirSync(dirname(outPath), { recursive: true });
    writeFileSync(outPath, JSON.stringify(json, null, 2));

    let fileStrings = 0;
    let fileChars = 0;
    for (const entry of json.entries) {
      const en = entry.strings?.en;
      if (en && en.trim()) {
        fileStrings++;
        fileChars += en.length;
      }
    }

    totalStrings += fileStrings;
    totalChars += fileChars;
    summary.push({ file: relPath, entries: json.entries.length, strings: fileStrings, chars: fileChars });
  } catch (err) {
    console.error(`ERROR: ${file}: ${err.message}`);
  }
}

console.log(`\n=== SUMMARY ===`);
console.log(`Files: ${files.length}`);
console.log(`Total English strings: ${totalStrings}`);
console.log(`Total characters: ${totalChars}`);
console.log(`\nPer file:`);
summary.sort((a, b) => b.chars - a.chars);
for (const s of summary) {
  console.log(`  ${s.strings} strings (${s.chars} chars) - ${s.file}`);
}
